File: 0013-roman-to-integer/0013-roman-to-integer.cpp
using namespace std;
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> mp;
        mp['I'] = 1;
        mp['V'] = 5;
        mp['X'] = 10;
        mp['L'] = 50;
        mp['C'] = 100;
        mp['D'] = 500;
        mp['M'] = 1000;
        string:: iterator it;
        it = s.end();
        int sum = *it;
        for (it = s.end()-1; it != s.begin()-1; it--) {
            if (mp[*it] < mp[*(it+1)]) {
                cout << "it:" << *it << endl;
                cout << "it+1:" << *(it+1) << endl;
                sum -= mp[*it];
                cout << "Map Value:" << mp[*it] << endl;
            }
            else {
                sum += mp[*it];
                cout << "Iterator:" << mp[*it] << endl;
            }
        }
        return sum;
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        map = {"I":1, "V":5, "X":10, "L":50, "C":100, "D": 500, "M":1000}
        integer = map[s[-1]]
        for i in range(len(s)-2, -1, -1):
            if map[s[i]] < map[s[i+1]]:
                integer -= map[s[i]]
            else:
                integer += map[s[i]]
        return integer